### ISSUE TITLE
Multivariate model parsing fails with custom response distribution

### ISSUE THREAD

**Issue by RaoulWolf** [paul-buerkner/brms#906]
***tl;dr** 
multivariate models (using mvbf) fail to parse when using custom response distributions (like beta binomial).*

I recently encountered a possible bug when using custom families in multivariate models.

Here's a regex based on the example in the `custom_family()` documentation, extending the beta binomial model with a lognormal model:
```
library(brms)
phi <- 0.7
n <- 300
z <- rnorm(n, sd = 0.2)
ntrials <- sample(1:10, n, replace = TRUE)
eta <- 1 + z
mu <- exp(eta) / (1 + exp(eta))
a <- mu * phi
b <- (1 - mu) * phi
p <- rbeta(n, a, b)
y <- rbinom(n, ntrials, p)
## The next two variables were added to setup the additional lognormal model
l <- rlnorm(n)
cat <- sample(c("a", "b"), n, replace = TRUE)

dat <- data.frame(y, z, ntrials, l, cat)
```

Next, the beta binomial distribution are defined using `custom_family()` and `stanvars()`:
```
beta_binomial2 <- custom_family(
  "beta_binomial2", dpars = c("mu", "phi"),
  links = c("logit", "log"), lb = c(NA, 0),
  type = "int", vars = "trials[n]"
)

stan_funs <- "
  real beta_binomial2_lpmf(int y, real mu, real phi, int N) {
    return beta_binomial_lpmf(y | N, mu * phi, (1 - mu) * phi);
  }
"

stanvars <- stanvar(scode = stan_funs, block = "functions")
```

The univariate beta binomial model parses without problems:
```
fit1 <- brm(y | trials(ntrials) ~ z, data = dat, 
            family = beta_binomial2, stanvars = stanvars)
```

Also when specifying different `phi`s based on the `cat` variable, the model parses:
```
fit2 <- brm(bf(y | trials(ntrials) ~ z, flist = list(phi ~ cat), family = beta_binomial2), 
            data = dat, stanvars = stanvars)
```

The univariate lognormal model parses as well; the `stanvars` parameter serves no purpose here (I assume?), but also doesn't seem to trigger any errors:
```
fit3 <- brm(bf(l ~ z, flist = list(sigma ~ cat), family = lognormal), 
            data = dat, stanvars = stanvars)
```

Things get problematic when combining the beta binomial model and the lognormal model in a bivariate model:
```
fit4 <- brm(mvbf(bf(formula = y | trials(ntrials) ~ z, flist = list(phi ~ cat), family = beta_binomial2),
                 bf(formula = l ~ z, flist = list(sigma ~ cat), family = lognormal)),
            data = dat, stanvars = stanvars)
```

The parsing of the model for `fit4` fails with the following error message:
```
SYNTAX ERROR, MESSAGE(S) FROM PARSER:
Variable "trials" does not exist.
 error in 'model4e142376876_file4e141d694474' at line 81, column 67
  -------------------------------------------------
    79:   if (!prior_only) {
    80:     for (n in 1:N_y) {
    81:       target += beta_binomial2_lpmf(Y_y[n] | mu_y[n], phi_y, trials[n]);
                                                                          ^
    82:     }
  -------------------------------------------------

Error in stanc(model_code = paste(program, collapse = "\n"), model_name = model_cppname,  : 
  failed to parse Stan model 'file4e141d694474' due to the above error.
```

I tried a couple of workarounds, but to no success (they all result in the same parsing error):
* replacing `"trials[n]"` with `"vint1[n]"` in the `custom_family()` specification (as suggested in the vignette) and subsequently using `y | vint(ntrials)` in the formula specification.
* Running the multivariate model without `flist = list(phi ~ cat)`.
* Using the current GitHub version of brms (2.12.11) instead of the current CRAN version (2.12.0).

Maybe there are some additional steps necessary when using custom response distributions in multivariate models, but I wasn't able to find any information on this.

**Comment by paul-buerkner** [comment]
Thank you for opening this issue and for providing a minimal reproducible example! This helped a lot in tracking down the error. There are multiple ways to fix it so I will think about it a bit today before I implement one of these fixes.

**Comment by RaoulWolf** [comment]
Thanks for the swift response! Really appreciate all the work and detailed care that goes into this package 👍 

By the way, it seems that *all* post processing examples of the ["Define Custom Response Distributions with brms" vignette](https://cran.r-project.org/web/packages/brms/vignettes/brms_customfamilies.html) are not working anymore. I get different error messages depending on the brms version (CRAN 2.12.0, GitHub 2.12.11). 

I will have to do post processing of a multivariate model with a custom response distribution eventually (`loo()`, `pp_check()`, ...), so this will pop up again; if you want I can open a separate issue.

**Comment by paul-buerkner** [comment]
I will check, thank you!

**Comment by paul-buerkner** [comment]
Both the issue with the multivariate model and the vignette should now be fixed.
